### PR TITLE
[PAN 1881] Add sonar workflow

### PR DESCRIPTION
**Pull request by juanmanuel-tirado**
## What type of PR is this? (check all applicable)

- [x] Refactor
- [x] Feature
- [ ] Bug Fix
- [x] Optimization
- [ ] Documentation Update

## Description
For PRs triggered from forked repos the `SONAR_SECRET` cannot be shared. To bypass this limitation and enforce the sonar analysis we trigger it using a `workflow_run` hook.

## Related Tickets & Documents

- Related Issue https://github.com/pantos-io/ci-workflows/pull/6

## QA Instructions

After completing the CI workflow, Sonar analysis must be triggered and the corresponding artifacts downloaded accordingly.

### CODE REVIEW

**Comment by juanmanuel-tirado** [comment]
The sonar job will fail because the new v1.0.3 is not yet moved to the main branch.

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -8,5 +8,4 @@ on:
 
 jobs:
   shared:
-    uses: pantos-io/ci-workflows/.github/workflows/python-ci.yml@84c18c1b5bf447e4a02f94c45c5266abab52bd55
-    secrets: inherit  
+    uses: pantos-io/ci-workflows/.github/workflows/python-ci.yml@v1.0.3
```
Comment (danut13) [inline .github/workflows/ci.yaml:11]: Here we should still reference `v1` and not `v1.0.3`. Then, in the `ci-workflows` repo we should tag the commit of `v1.0.3` also with `v1`

According to https://github.com/actions/toolkit/blob/main/docs/action-versioning.md:
> Binding to a major version is the latest of that major version ( e.g. v1 == "1.*" )

Comment (juanmanuel-tirado) [reply .github/workflows/ci.yaml:11]: I don't like the idea of not setting specific versions specially when we're playing with the current code. The good thing is that we will not need new changes in `common` for new releases of the CI 🤔 I will change it for the sake of convenience. Thanks for the point.

**Comment by juanmanuel-tirado** [comment]
I will close this PR in favour of #17

**Pull request closed without merging**